library/src/lib/button-split/button-split.component.ts

```diff
@@ -0,0 +1,128 @@
+import { Component, ContentChild, ElementRef, EventEmitter, Input, Output, TemplateRef } from '@angular/core';
+import { AbstractFdNgxClass } from '../utils/abstract-fd-ngx-class';
+import { ButtonSplitActionTitle } from './button-split-utils/button-split.directives';
+
+/**
+ * Split Button component, used to enhance standard HTML button and add possibility to put some dropdown with
+ * additional options.
+ *
+ * ```html
+ *    <fd-button-split>
+ *        Action Button
+ *        <div fd-button-split-menu>
+ *            <fd-menu>
+ *                <ul fd-menu-list>
+ *                    <li fd-menu-item>
+ *                       <a [routerLink]="'/'">option</a>
+ *                    </li>
+ *                    <li fd-menu-item>
+ *                       <a [routerLink]="'/'">option2</a>
+ *                    </li>
+ *                </ul>
+ *            </fd-menu>
+ *        </div>
+ *    </fd-button-split>
+ * ```
+ */
+@Component({
+    // TODO to be discussed
+    // tslint:disable-next-line:directive-selector
+    selector: 'fd-button-split',
+    templateUrl: 'button-split.component.html'
+})
+export class ButtonSplitComponent extends AbstractFdNgxClass {
```

Comment:
If it's not needed, better to not extend AbstractFdNgxClass. I'm currently looking to progressively phase this out as we go